src/MultiFormats/Cryptography/Keccak.cs

```diff
@@ -0,0 +1,152 @@
+            Initialize();
+            HashSizeValue = hashBitLength;
+            switch (hashBitLength)
+            {
+                case 224:
+                    KeccakR = 1152;
+                    break;
+                case 256:
+                    KeccakR = 1088;
+                    break;
+                case 384:
+                    KeccakR = 832;
+                    break;
+                case 512:
+                    KeccakR = 576;
+                    break;
+            }
+
+            RoundConstants = new ulong[]
+            {
+                0x0000000000000001UL,
+                0x0000000000008082UL,
+                0x800000000000808aUL,
+                0x8000000080008000UL,
+                0x000000000000808bUL,
+                0x0000000080000001UL,
+                0x8000000080008081UL,
+                0x8000000000008009UL,
+                0x000000000000008aUL,
+                0x0000000000000088UL,
+                0x0000000080008009UL,
+                0x000000008000000aUL,
+                0x000000008000808bUL,
+                0x800000000000008bUL,
+                0x8000000000008089UL,
+                0x8000000000008003UL,
+                0x8000000000008002UL,
+                0x8000000000000080UL,
+                0x000000000000800aUL,
+                0x800000008000000aUL,
+                0x8000000080008081UL,
+                0x8000000000008080UL,
+                0x0000000080000001UL,
+                0x8000000080008008UL
+            };
+        }
+
+        protected ulong Rol(ulong a, int offset)
+        {
+            return (a << (offset % KeccakLaneSizeInBits)) ^
+                (a >> (KeccakLaneSizeInBits - offset % KeccakLaneSizeInBits));
+        }
+
+        protected void AddToBuffer(byte[] array, ref int offset, ref int count)
+        {
+            var amount = Math.Min(count, buffer.Length - buffLength);
+            Buffer.BlockCopy(array, offset, buffer, buffLength, amount);
+            offset += amount;
+            buffLength += amount;
+            count -= amount;
+        }
+
+        public override byte[] Hash => HashValue;
+
+        public override int HashSize => HashSizeValue;
+
+        #endregion
+
+        public override void Initialize()
+        {
+            buffLength = 0;
+            state = new ulong[5 * 5]; //1600 bits
+            HashValue = null;
+        }
+
+        protected override void HashCore(byte[] array, int ibStart, int cbSize)
+        {
+            if (array == null)
+                throw new ArgumentNullException("array");
+            if (ibStart < 0)
+                throw new ArgumentOutOfRangeException("ibStart");
+            if (cbSize > array.Length)
+                throw new ArgumentOutOfRangeException("cbSize");
+            if (ibStart + cbSize > array.Length)
+                throw new ArgumentOutOfRangeException("ibStart or cbSize");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name 'ibStart or cbSize' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)